vault/identity_store_util.go

```diff
@@ -931,6 +931,59 @@ func (i *IdentityStore) sanitizeAndUpsertGroup(ctx context.Context, group *ident
 	defer txn.Abort()
 
 	memberGroupIDs = strutil.RemoveDuplicates(memberGroupIDs, false)
+
+	// Remove ParentGroupID from removed GroupMembers
+	// Get the current MemberGroups IDs for this group
+	var currentMemberGroupIDs []string
+	currentMemberGroups, err := i.MemDBGroupsByParentGroupID(group.ID, false)
+	if err != nil {
+		return err
+	}
+	for _, currentMemberGroup := range currentMemberGroups {
+		currentMemberGroupIDs = append(currentMemberGroupIDs, currentMemberGroup.ID)
+	}
+
+	// Check if current MemberGroups should be removed
+	for _, currentMemberGroupID := range currentMemberGroupIDs {
+		currentMemberGroup, err := i.MemDBGroupByID(currentMemberGroupID, true)
+		if err != nil {
+			return err
+		}
+		if currentMemberGroup == nil {
+			return fmt.Errorf("invalid member group ID %q", currentMemberGroupID)
+		}
+
+		removed := true
+		for _, memberGroupID := range memberGroupIDs {
+			if memberGroupID == currentMemberGroupID {
+				removed = false
+				break
+			}
+		}
+
+		// Remove ParentGroup Entry for this group from removed Group
+		if removed {
+			for i, v := range currentMemberGroup.ParentGroupIDs {
```

Comment:
This `for` loop can be substituted with `currentMemberGroup.ParentGroupIDs = strutil.StrListDelete(currentMemberGroup.ParentGroupIDs, group.ID)`.

Reply:
thx, I changed this